spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateV3SpecTest.java

```diff
@@ -17,121 +17,493 @@
+        assertEquals(6400, archRetContent.get("metadata_expire").asInt());
+        assertThat(archRetContent.get("required_tags")).hasSize(2);
+        assertThat(archRetContent.get("arches")).hasSize(2);
+        String arches = archRetContent.get("arches").toString();
+        assert arches.contains("i386");
+        assert arches.contains("x86_64");
+
+        consumerApi.unbindAll(system.getUuid());
+    }
+
+    @Test
+    @DisplayName("verify branding info is correct in json blob")
+    public void verifyBrandingInTheBlob() throws Exception {
+        BrandingDTO branding = null;
+        ProductDTO mktProduct = null;
+        ProductDTO engProduct = null;
+        if (CandlepinMode.isHosted()) {
+            engProduct = Products.randomEng()
+                .name("engineering_product_name");
+            engProduct = hostedTestApi.createProduct(engProduct);
+            branding = Branding.random("Super Branded Name")
+                .type("Some Type")
+                .productId(engProduct.getId());
+            mktProduct = Products.randomSKU()
+                .name("marketing_product_name")
+                .branding(Set.of(branding))
+                .providedProducts(Set.of(engProduct));
+            mktProduct = hostedTestApi.createProduct(mktProduct);
+            subscription = Subscriptions.random(owner, mktProduct)
+                .contractNumber("12345")
+                .accountNumber("6789")
+                .orderNumber("order1")
+                .providedProducts(Set.of(engProduct));
+            subscription = hostedTestApi.createSubscription(subscription);
+            AsyncJobStatusDTO refresh = ownerApi.refreshPools(owner.getKey(), false);
+            if (refresh != null) {
+                client.jobs().waitForJob(refresh.getId());
+            }
+        }
+        else {
+            engProduct = Products.randomEng()
+                .name("engineering_product_name");
+            engProduct = ownerProductApi.createProductByOwner(owner.getKey(), engProduct);
+            branding = Branding.random("Super Branded Name")
+                .type("Some Type")
+                .productId(engProduct.getId());
+            mktProduct = Products.randomSKU()
+                .name("marketing_product_name")
+                .branding(Set.of(branding))
+                .providedProducts(Set.of(engProduct));
+            mktProduct = ownerProductApi.createProductByOwner(owner.getKey(), mktProduct);
+            ProvidedProductDTO engProvidedProduct = (new ProvidedProductDTO())
+                .productName(engProduct.getName())
+                .productId(engProduct.getId());
+            pool = Pools.random(mktProduct)
+                .contractNumber("12345")
+                .accountNumber("6789")
+                .orderNumber("order1")
+                .providedProducts(Set.of(engProvidedProduct));
+            pool = ownerApi.createPool(owner.getKey(), pool);
+            assertThat(pool.getBranding()).hasSize(1);
```

Comment:
shouldn't this be asserted outside the if/else to make sure it's checked for both hosted/standalone?